### ISSUE TITLE
something ? HIP / SPI pin conflict does not fail gracefully

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#2108]
HW_CI drama points at issues due to hip and/or SD threads using pins no longer allocated to those

open question is why functional tests do not fail with a nice and explicit error message

part of #2063 probably relayed to Rover config

**Comment by rusefillc** [comment]
i.e. why is the HW_CI failure sporadic and not permanent?

**Comment by rusefillc** [comment]
actually HIP is pretty disabled on Frankenso? this is all NOT about HIP SPI?

```
#ifndef EFI_HIP_9011
// disabling for now - DMA conflict with SPI1
#define EFI_HIP_9011 FALSE
#endif
```

**Comment by rusefillc** [comment]
^^^ would be build 2072

**Comment by rusefillc** [comment]
^^^ is 2076

**Comment by rusefillc** [comment]
^^^ 2081

**Comment by rusefillc** [comment]
http://rusefici.freeddns.org:23456/job/integration_and_primary_bundle/2090/console has failed before https://github.com/rusefi/rusefi/commit/bc2b5c71e60c430f19ffe2097c1fcdb2f0b6d875

maybe this has NOTHING to do with rover? maybe need more runs to see the issue?

**Comment by rusefillc** [comment]
2098 for https://github.com/rusefi/rusefi/commit/bc2b5c71e60c430f19ffe2097c1fcdb2f0b6d875

**Comment by rusefillc** [comment]
got 50 good runs overnight

![image](https://user-images.githubusercontent.com/48498823/102716110-bb0b8c80-42a7-11eb-8a34-4ecb2fb45a21.png)

**Comment by rusefillc** [comment]
http://rusefici.freeddns.org:23456/job/integration_and_primary_bundle/2158/ returns extremely limited rover test

**Comment by rusefillc** [comment]
^^^ still fails
http://rusefici.freeddns.org:23456/job/integration_and_primary_bundle/2173/ even less Rover config

**Comment by rusefillc** [comment]
http://rusefici.freeddns.org:23456/job/integration_and_primary_bundle/2182/ even less, LESS Rover config

**Comment by rusefillc** [comment]
http://rusefici.freeddns.org:23456/job/integration_and_primary_bundle/2194/ has a fix on java side - we skip one engine sniffer frame after change of engine type

Now need 50 more runs overnight to know for sure

**Comment by rusefillc** [comment]
this was a legit firmware bug which seems to be solved by #2109
